### PR TITLE
RequestInfoResolver: restrict special project logic to project API groups

### PR DESCRIPTION
**Pull request by sttts**
Fixes https://github.com/openshift/origin/issues/20251.

### CODE REVIEW

File: pkg/authorization/authorizer/project_request_info_resolver.go [hunk 1]
```diff
@@ -3,6 +3,7 @@ package authorizer
 import (
 	"net/http"
 
+	"github.com/openshift/origin/pkg/project/apis/project"
```
Comment (enj) [inline pkg/authorization/authorizer/project_request_info_resolver.go]: Can you move the import into openshift specific section.

File: pkg/authorization/authorizer/project_request_info_resolver.go [hunk 2]
```diff
@@ -24,7 +25,7 @@ func (a *projectRequestInfoResolver) NewRequestInfo(req *http.Request) (*apirequ
 	}
 
 	// if the resource is projects, we need to set the namespace to the value of the name.
-	if (requestInfo.Resource == "projects") && (len(requestInfo.Name) > 0) {
+	if (requestInfo.APIGroup == "" || requestInfo.APIGroup == project.GroupName) && requestInfo.Resource == "projects" && len(requestInfo.Name) > 0 {
```
Comment (enj) [inline pkg/authorization/authorizer/project_request_info_resolver.go]: Don't we prefer the `len(...) == 0` check?

**Comment by enj** [comment]
/hold

Just some nits if you care to address.

**Comment by dbazhal** [comment]
@sttts any help required? would be nice to see this fix merged

**Pull request merged**